translators/Weixin.js

```diff
@@ -0,0 +1,82 @@
+{
+	"translatorID": "5a325508-cb60-42c3-8b0f-d4e3c6441058",
+	"label": "Weixin",
+	"creator": "Fushan Wen",
+	"target": "^https?://([^/]+\\.)?mp\\.weixin\\.qq\\.com",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcs",
+	"lastUpdated": "2020-11-20 21:37:05"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+	This file is part of Zotero.
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+	***** END LICENSE BLOCK *****
+*/
+
+function getMetaContent(doc, property) {
+	return ZU.xpath(doc, "/html/head/meta[@property='" + property + "']")[0].content.trim();
+}
+
+function getArticleDate(doc) {
+	var script_list = ZU.xpath(doc, "//script");
+	for (var sc of script_list) {
+		if (sc.innerText.includes("今天")) {
+			Zotero.debug("Good date");
+			return sc.innerText.match(/s=\"(.+)\"/)[1].trim();
+		}
+	}
+	return "";
+}
+
+function getArticleCreator(doc) {
+	var creatorName = getMetaContent(doc, "og:article:author");
+	var accountName = ZU.xpath(doc, "//a[@id='js_name']")[0].innerText.trim();
+	if (!creatorName.length) {
+		return [{lastName: accountName, creatorType: "author", fieldMode: 1}];
+	}
+	return [{lastName: creatorName, creatorType: "author", fieldMode: 1}, {lastName: accountName, creatorType: "author", fieldMode: 1}];
+}
+
+function scrape(doc, url) {
+	var item = new Zotero.Item("webpage");
+	item.title = getMetaContent(doc, "og:title");
+	item.websiteTitle = getMetaContent(doc, "og:site_name");
+	item.url = getMetaContent(doc, "og:url");
+	item.abstractNote = getMetaContent(doc, "og:description");
+	item.creators = getArticleCreator(doc);
+	item.date = getArticleDate(doc);
+	item.accessDate = new Date().toISOString().slice(0, 10);
+	item.extra = ZU.xpath(doc, "//div[@id='js_content']")[0].innerText.trim().replace(/[\r\n]+/g, "\n");
+	item.complete();
+}
+
+function detectWeb(doc, url) {
+	var title = getMetaContent(doc, "og:title");
+	if (title.length) {
+		return "webpage";
+	}
+	else {
+		return false;
+	}
+}
```

Comment:
```suggestion
function detectWeb(doc, url) {
    if (doc.title) {
        return "webpage";
    } else {
        return false;
    }
}
```

没明白这个判断的意义：

* “一个已发布的公众号文章的标题为空”应该是件很奇怪的事情。
* 为什么取 OG title，而不是 HTML title？

---

我感觉判 OG type 应该更有意义。**以下是我猜的**，因为我没见过不是文章的公众号文章。

```js
/**
 * @param {Document} doc
 * @param {string} url
 */
function detectWeb(doc, url) {
    /**
     * @type {HTMLMetaElement | null}
     */
    const ogType = doc.head.querySelector('meta[property="og:type"]');

    if (ogType && ogType.content) {
        switch (ogType.content.split(".")[0]) {
            case "article":
                return "document";

            case "video":
                return "videoRecording";

            default:
                return "webpage";
        }
    } else {
        return false;
    }
}
```

不过，那样的话，`doWeb()` 也得跟着改。

Reply:
受教了，下周我弄下。

Reply:
同学加油呀，感谢@Lemmingh